### PR TITLE
fix: wallet connect - sake.lido.fi dApp - staking transaction gets stuck

### LINKED ISSUE #16096: Wallet connect - Sake.lido.fi dApp - Staking transaction gets stuck
# Bug Report

## Steps to reproduce

1. Log into an account with funds
2. Connect the wallet to https://stake.lido.fi/
3. Stake some eth
4. Sign the approval on the desktop app

#### Expected behavior
Transaction to succeed

#### Actual behavior
Screenshot 2024-08-13 at 12 22 21 PM

Lido website shows: "You are staking 0.0002 ETH"
Etherscan: https://etherscan.io/tx/0x3ef063390129dde22b802941279dd960ea7321b2cffc599a5355ad4114e0a38c
Screenshot 2024-08-13 at 12 26 49 PM

Desktop app shows:
Screenshot 2024-08-13 at 12 22 04 PM
Screenshot 2024-08-13 at 12 21 52 PM

Attaching console log of test account
[Stake lido app cannot stake test035b.txt](https://github.com/user-attachments/files/16601841/Stake.lido.app.cannot.stake.test035b.txt)

### Additional Information

- Status desktop version: 2.30 rc-7
- Operating System: Mac OS Sonoma

### PR DESCRIPTION
**Pull request by saledjenic**
Placed tx:
- https://etherscan.io/tx/0xa60423657d8d27a1754ce857282a5d22a5ba3bc53ea829acdccbf89ffabe6bc7

image

Fixes #16096

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml [hunk 1]
```diff
@@ -484,13 +484,11 @@ SQUtils.QObject {
 
             // Beware, the tx values are standard blockchain hex big number values; the fees values are nim's float64 values, hence the complex conversions
             if (!!tx.maxFeePerGas && !!tx.maxPriorityFeePerGas) {
-                let maxFeePerGasDec = root.store.hexToDec(tx.maxFeePerGas)
-                const gasPriceInWei = BigOps.fromString(maxFeePerGasDec)
+                maxFeePerGas = hexToGwei(tx.maxFeePerGas)
+                maxPriorityFeePerGas = hexToGwei(tx.maxPriorityFeePerGas)
+
+                // TODO: check why we need to set gasPrice here and why if it's not checked we cannot send the tx and fees are unknown????
                 gasPrice = hexToGwei(tx.maxFeePerGas)
```
Comment (saledjenic) [inline ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:491]: Not sure why `gasPrice` is used in this context (refer to TODO in the line above)?

File: src/app_service/common/json_utils.nim [hunk 2]
```diff
@@ -36,15 +36,15 @@ template getProp(obj: JsonNode, prop: string, value: var typedesc[uint64]): bool
 
 template getProp(obj: JsonNode, prop: string, value: var typedesc[string]): bool =
   var success = false
-  if (obj.kind == JObject and obj.contains(prop)):
+  if (obj.kind == JObject and obj.contains(prop) and obj[prop].kind == JString):
```
Comment (saledjenic) [inline src/app_service/common/json_utils.nim:39]: All functions in this file should have this check `and obj[prop].kind == ....` but when I did that I couldn't start the app (got crash) which undoubtedly means that in some places we have incorrect type usage or receiving incorrect json data. That should be fixed in another PR.

**Review by virginiabalducci — APPROVED** [review]
the staking tx works on my end

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml [hunk 3]
```diff
@@ -484,13 +484,11 @@ SQUtils.QObject {
 
             // Beware, the tx values are standard blockchain hex big number values; the fees values are nim's float64 values, hence the complex conversions
             if (!!tx.maxFeePerGas && !!tx.maxPriorityFeePerGas) {
```
Comment (alaibe) [inline ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:486]: as we are ine post eip1559 case, both are present, we do not need to set gas price i believe

**Review by alaibe — APPROVED** [review]

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml (hunk 3 above)
Comment (saledjenic) [reply ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:486]: exactly, that was my question about, [here](https://github.com/status-im/status-desktop/pull/16348#discussion_r1761565396).

**Review by alexjba — APPROVED** [review]
LGTM!
